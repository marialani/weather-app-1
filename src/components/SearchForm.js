import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";

const SearchForm = ({ searchText, setSearchText, onSubmit }) => {
  const handleInputChange = (event) => {
    setSearchText(event.target.value);
  };

  return (
    <div className="search-form">
      <input
        placeholder="Enter city name"
        className="search-form_input"
        type="text"
        onChange={handleInputChange}
        value={searchText}
      />
      <button type="submit" onClick={onSubmit} className="search-form_button">
        Search
      </button>
    </div>
  );
};

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default SearchForm;
