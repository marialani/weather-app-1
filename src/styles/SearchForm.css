.search-form {
  display: flex;
  justify-content: center;
  margin-bottom: 4rem;
}

.search-form_input {
  margin-right: 2rem;
  border: #000a47;
  border-style: groove;
}

.search-form_button {
  display: inline-block;
  outline: none;
  cursor: pointer;
  font-weight: 600;
  border-radius: 3px;
  padding: 12px 24px;
  border: 0;
  color: #fff;
  background: #000a47;
  line-height: 1.15;
  font-size: 16px;
}
