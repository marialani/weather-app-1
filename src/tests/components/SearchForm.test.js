import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "stubCity",
    setSearchText: () => {},
    onSubmit: () => {},
  };

  it("renders", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the search button text", () => {
    const { getByText } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(getByText("Search")).toHaveClass("search-form_button");
  });
});
