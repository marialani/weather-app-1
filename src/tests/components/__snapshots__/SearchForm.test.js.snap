// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      type="text"
      value="stubCity"
    />
    <button
      class="search-form_button"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
